vnext/ReactUWP/Base/UwpReactInstance.cpp

```diff
@@ -243,6 +243,20 @@ void UwpReactInstance::Start(const std::shared_ptr<IReactInstance>& spThis, cons
     devSettings->jsExceptionCallback = std::move(settings.JsExceptionCallback);
     devSettings->useJITCompilation = settings.EnableJITCompilation;
     devSettings->debugHost = settings.DebugHost;
+    devSettings->useCustomRootPath = settings.UseCustomRootPath;
```

Comment:
I did some work for Office to allow specifying the different parts of the source bundle url (host/port/path/extension). For Excel Custom Functions, we needed to alter the path. It would be great to see about normalizing those changes in here. This is primarily the name for the dev settings, and having it compose the source bundle url when it goes to request the bundle from the packager.

Reply:
OK, there's "devSettings->debugBundlePath" now. I didn't modify devSettings with the other components because of the work to refactor existing code to support a different host, port, or extension.  Looking at this now, I think your usage of the bundle path might be different than getting the bundle from the packager when debugging.

Reply:
Did you have a particular standard in mind? Maybe something like this?

```suggestion
    devSettings->useCustomRootPath = settings.UseCustomRootPath;
    devSettings->bundleHost = settings.BundleHost;
    devSettings->bundlePort = settings.BundlePort;
    devSettings->bundlePath = settings.BundlePath;
    devSettings->bundleExtension = settings.BundleExtension;
```

Could populate these with the default values when UseCustomRootPath is set to false. Also, I suppose the other issue would be whether or not the bundle name should continue to be passed into OInstance separately or not.

Reply:
Ah, sorry for the confusion. Yes, this is specifically for non-debug scenarios, where the bundle is loaded from disk.

Reply:
I don't think you need the separate bool "useCustomRootPath". Yes, the host/port/path/extension allows specifying a portion but falling back to the default for the unspecified parts. If nothing it specified directly, then it would use the entire default that exists now.

I'm not familiar with exactly how the m_bundleRootPath is used now. I believe this is for loading the bundle from the app resources. At first I was confusing this with web debugging where there is a separate url used to load the bundle from the packager. Sorry!

If m_bundleRootPath is the path only, then I don't think there is a need to allow overriding host/port/extension. Better to keep it simple.

Reply:
The intent of having the separate bool was to differentiate between the developer leaving a value intentionally blank versus blank due to lack of use. That being said, a blank value there would probably not work anyways based on how the bundle is being loaded, so maybe that's overkill. I can just add logic to replace the root path with "ms-appx://Bundle/" if it's passed in as blank, if you think that would make more sense.

m_bundleRootPath is just the path (same as in DevSettings), and is intended to be used for components that don't have access to the dev settings but still need to access the bundle. As of right now, that applies specifically to ReactImage, which can access the path through the new helper.

Reply:
Yes, I think it makes more sense to just have one setting to override rather than require two.